replica/exceptions.cc

```diff
@@ -24,6 +24,8 @@ exception_variant try_encode_replica_exception(std::exception_ptr eptr) {
         return rate_limit_exception();
     } catch (const stale_topology_exception& e) {
         return e;
+    } catch (abort_requested_exception&) {
+        return abort_requested_exception();
```

Comment:
BTW, it looks like the `try_encode_replica_exception` function should be updated to use `utils::try_catch` and not throw. This is out of scope of this PR so no need to address it here, but it could be done in a follow up.

Reply:
Also why do we return `no_exception{}` in the catch-all clause? Doesn't seem right.